docs/src/content/docs/guides/authoring-content.md

```diff
@@ -202,9 +202,140 @@ var fun = function lang(l) {
 ```
 ````
 
-```md
-Long, single-line code blocks should not wrap. They should horizontally scroll if they are too long. This line should be long enough to demonstrate this.
-```
+### Expressive Code features
+
+Starlight uses [Expressive Code](https://github.com/expressive-code/expressive-code/tree/main/packages/astro-expressive-code) to extend formatting possibilities for code blocks.
+Expressive Code’s text markers and window frames plugins are enabled by default.
+Code block rendering can be configured using Starlight’s [`expressiveCode` configuration option](/reference/configuration/#expressivecode).
+
+#### Text markers
+
+You can highlight specific lines or parts of your code blocks using text markers.
+There are three styles of marker: a default neutral colored highlight, a green highlight indicating code that was inserted, and a red highlight indicating code that was deleted.
+
+- [Mark entire lines & line ranges](https://github.com/expressive-code/expressive-code/blob/main/packages/%40expressive-code/plugin-text-markers/README.md#marking-entire-lines--line-ranges):
+
+  ```js {2-3}
+  function demo() {
+    // This line (#2) and the next one are highlighted
+    return 'This is line #3 of this snippet';
+  }
+  ```
+
+  ````md
+  ```js {2-3}
+  function demo() {
+    // This line (#2) and the next one are highlighted
+    return 'This is line #3 of this snippet';
+  }
+  ```
+  ````
+
+- [Mark selections of text](https://github.com/expressive-code/expressive-code/blob/main/packages/%40expressive-code/plugin-text-markers/README.md#marking-entire-lines--line-ranges):
+
+  ```js "Individual terms" /Even.*supported/
+  // Individual terms can be highlighted, too
+  function demo() {
+    return 'Even regular expressions are supported';
+  }
+  ```
+
+  ````md
+  ```js "Individual terms" /Even.*supported/
+  // Individual terms can be highlighted, too
+  function demo() {
+    return 'Even regular expressions are supported';
+  }
+  ```
+  ````
+
+- [Mark text as inserted or deleted with `ins` or `del`](https://github.com/expressive-code/expressive-code/blob/main/packages/%40expressive-code/plugin-text-markers/README.md#selecting-marker-types-mark-ins-del):
+
+  ```js "return true;" ins="inserted" del="deleted"
+  function demo() {
+    console.log('These are inserted and deleted marker types');
+    // The return statement uses the default marker type
+    return true;
+  }
+  ```
+
+  ````md
+  ```js "return true;" ins="inserted" del="deleted"
+  function demo() {
+    console.log('These are inserted and deleted marker types');
+    // The return statement uses the default marker type
+    return true;
+  }
+  ```
+  ````
+
+- [Combine syntax highlighting with `diff`-like syntax](https://github.com/expressive-code/expressive-code/blob/main/packages/%40expressive-code/plugin-text-markers/README.md#combining-syntax-highlighting-with-diff-like-syntax):
```

Comment:
```suggestion
- [Combine language-based syntax highlighting with `diff`-like syntax](https://github.com/expressive-code/expressive-code/blob/main/packages/%40expressive-code/plugin-text-markers/README.md#combining-syntax-highlighting-with-diff-like-syntax):
```

OOh, question, does this mean that for Starlight, we do NOT do ins/del for individual lines? I know this is the solution we use for standard Markdown, but our normal `ins={2-4}` should work in Starlight, right? Does this mean you are suggesting this notation (for universal compatibility?)

If so, I might further explain this not as the first resort, but as a way to write code samples in such a way that they'll render in non-Expressive Code environments as diffs, but in your Starlight site, they can take advantage additionally of language highlighting.  (In part, because our code samples in Docs are typically going to use the `ins={2-4}` notation, and so the docs here should maybe have an example of that?)

Reply:
I believe both diff-style and `ins={...}` should work! So it’s a matter of taste which you choose (although I think the diff-style is probably preferable in most cases for a couple of reasons:

1. Diff highlighting works everywhere (same reason we were using it in Astro integration READMEs).
2. You don’t need to count and maintain line numbers. If you use `+`/`-` at the start of lines, adding or moving parts of the code doesn’t require you to manually update the `ins`/`del` ranges.

Reply:
They do both still work, yes. Nothing has been removed. I just tried to limit the amount of EC-focused examples included in the Starlight docs to a reasonable amount and therefore decided to just show an example of the diff-like syntax for the reasons Chris mentioned.

Reply:
Yes, I think once Expressive Code has its own docs to show these off, we may even be able to reduce the examples in favour of linking there.